src/sentry/integrations/on_call/metrics.py

```diff
@@ -0,0 +1,59 @@
+from collections.abc import Mapping
+from enum import Enum
+from typing import Any
+
+from attr import dataclass
+
+from sentry.integrations.opsgenie.spec import OpsgenieOnCallSpec
+from sentry.integrations.utils.metrics import EventLifecycleMetric, EventLifecycleOutcome
+from sentry.models.organization import Organization
+from sentry.organizations.services.organization import RpcOrganization
+from sentry.users.models import User
+from sentry.users.services.user import RpcUser
+
+
+class OnCallInteractionType(Enum):
+    """
+    A way in which a user can interact with Sentry through an on-call app.
+    """
+
+    # TODO: add the interactions
+    # General interactions
+    ADD_KEY = "ADD_KEY"
```

Comment:
More accurately, for Opsgenie, this is "update config." Probably going to change this.